### DISCUSSION TITLE
Extend Hit highlighting summary length

### DISCUSSION THREAD

**Discussion by ferrarirosso** [microsoft-search/pnp-modern-search#2005] [Q&A]
We are using pnp-modern-search and the standard summary results.

We have noticed, that for some results, e.g. when the hit is more at the end of a document or page, there is no hit highlighting which is shown in the results, so we know the document or page contains the search words but have no context in the result summary.

Is there any way to configure this within the pnp modern search, specifically to extend hit highlighting on a biggest part of the result object ?

Thanks

**Comment by wobba** [comment]
Depending on the document size this is expected and an API behavior we cannot control. Increasing the size of characters you want for the summary won't have an impact on this, and not a setting we plan to add.

**Comment by ferrarirosso** [comment]
Thanks
There are two aspects here :
1. Extending the Highlighted Summary which is shown in the results.
2. Extending the size of the document which is used for hit highlighting.

For 1 I did find some hint here [Increase Hit Highlighted Summary](https://answers.microsoft.com/en-us/msoffice/forum/all/how-to-increase-hithighlightedsummary-length-from/64d9380f-fa5c-4e8c-a503-893a895bbf8e?msclkid=be32c9edba7f11ecab99642b531e4458) but that's not what I am searching for.
For 2, this is something which is done automatically and which I cannot influenec, correct, whether in any API, nor through the PnP modern search ?
